gamemodes/terrortown/gamemode/server/sv_eventpopup.lua

```diff
@@ -0,0 +1,48 @@
+util.AddNetworkString("ttt2_eventpopup")
+
+EPOP = EPOP or {}
+
+---
+-- Adds a translated popup message to the @{EPOP}
+-- @param table|Player plys A table of player that should receive this popup, broadcasts it if nil
+-- @param string|table title The title of the popup that will be displayed in large letters (can be a table with `text` and `color` attribute)
+-- @param [opt]string|table subtitle An optional description that will be displayed below the title (can be a table with `text` and `color` attribute)
+-- @param [default=4]number displayTime The render duration of the popup
+-- @param [default=true]boolean blocking If this is false, this message gets instantly replaced if a new message is added
+-- @realm server
+function EPOP:AddMessage(plys, title, subtitle, displayTime, blocking)
+	net.Start("ttt2_eventpopup")
+
+	if title or isstring(title) then
+		net.WriteBool(false)
+		net.WriteString(title or "")
+	elseif not IsValid(title.color) then
+		net.WriteBool(false)
+		net.WriteString(title.text or "")
+	else
+		net.WriteBool(true)
+		net.WriteString(title.text or "")
```

Comment:
couldn't this be simplified? In any case you write `title.text or ""`

Reply:
And is this `or ""` still needed?

Reply:
Above same as discussed before, you could use `title = title or ""` to reduce checks

Reply:
Maybe I don't understand what you mean, but what's the benefit here? Only one of the two `title.text or ""` is executed since they are in different parts of the `if`.

Reply:
You don't need to write `or ""` if the given input can't be nil. That what I'm asking. You do it in many parts, it looks like you don't know about the states of your own variables, the code looks inconsistent

Reply:
No, the user can call the function however he likes. Therefore I don't know if this value is set or not

Reply:
Hmm you are right. But if no title and no subtitle is given, you even don't need networking anything... Additionally, the title check above (as already discussed before) can be improved anyways